src/main.rs

```diff
@@ -1,12 +1,60 @@
 use std::{env, fs};
 
 
+
+fn find_file(path: String, file: String) {
+    let mut queue = Vec::new();
+
+    queue.push(path.clone());
+
+
+    while !queue.is_empty() {
+
+        let path = queue.remove(0);
+
+        let paths =  fs::read_dir(path).expect("Error");
+
+        for p in paths {
+            let p_unwrapped = p.unwrap();
+            if p_unwrapped.path().is_file() && p_unwrapped.file_name().to_str().expect("") == file {
+                println!("Found full path: {:?}", p_unwrapped.path().to_str());
+                return;
+            }
+            else if !p_unwrapped.path().is_file() {
+                queue.push(p_unwrapped.path().display().to_string());
+            }
+        }
+    }
+
+
+    println!("File not found");
+
+}
+
+
+fn recursive_output(p: &str) {
```

Comment:
Please, don't  use references (you'll have a chance to use them in the next HW). I want you to use move semantic and copy/clone.